internal/monitors/collectd/solr/solr.go

```diff
@@ -0,0 +1,78 @@
+package solr
+
+import (
+	"github.com/signalfx/signalfx-agent/internal/core/config"
+	"github.com/signalfx/signalfx-agent/internal/monitors"
+	"github.com/signalfx/signalfx-agent/internal/monitors/collectd"
+	"github.com/signalfx/signalfx-agent/internal/monitors/collectd/python"
+	"github.com/signalfx/signalfx-agent/internal/monitors/pyrunner"
+)
+
+const monitorType = "collectd/solr"
+
+// MONITOR(collectd/solr): Monitors Solr instances.
+//
+// See https://github.com/signalfx/collectd-solr and
+// https://github.com/signalfx/integrations/tree/master/collectd-solr
+
+func init() {
+	monitors.Register(monitorType, func() interface{} {
+		return &Monitor{
+			python.PyMonitor{
+				MonitorCore: pyrunner.New("sfxcollectd"),
+			},
+		}
+	}, &Config{})
+}
+
+// Config is the monitor-specific config with the generic config embedded
+type Config struct {
+	config.MonitorConfig `yaml:",inline" acceptsEndpoints:"true"`
+	pyConf               *python.Config
+	Host                 string `yaml:"host" validate:"required"`
+	Port                 uint16 `yaml:"port" validate:"required"`
+	// Cluster name of this solr cluster.
+	Cluster string `yaml:"cluster"`
+	// EnhancedMetrics boolean to indicate whether stats from /metrics are needed
+	EnhancedMetrics *bool `yaml:"enhancedMetrics" default:"false"`
+	// IncludeMetric metric name from the /admin/metrics endpoint to include (valid when EnhancedMetrics is "false")
+	IncludeMetric string `yaml:"includeMetric"`
+	// ExcludeMetric metric name from the /admin/metrics endpoint to exclude (valid when EnhancedMetrics is "true")
+	ExcludeMetric string `yaml:"excludeMetric"`
+	// Dimension space-separated key-value pair for a user-defined dimension
+	Dimension string `yaml:"dimension"`
```

Comment:
This is unnecessary since the agent has the ability to add dimensions via the `extraDimensions` key that is inherent to every monitor.  Remove this and don't pass it to the plugin.

Reply:
Ok